### PR TITLE
Suppress low-confidence SVs in `strict` mode

### PR DESCRIPTION
**Pull request by pradh**
Currently, we have a low-confidence threshold (below which we print a message) of `0.7`, and a drop threshold (below which SVs are dropped) of `0.5` for SVs.

The main change in this PR is, for`strict` mode, we increase the drop threshold to `0.7`.  This is done by plumbing in the "threshold" value to the NL server, depending on the setting of `mode=strict`.

Importantly, the `strict` mode is:
* supported only for `api/data` API that nodejs uses
* incompatible with LLM.

Concretely, setting of `mode=strict` implies:  `sv_threshold=0.7`, `detector=heuristic`, `use_default_place=False`

Additionally,
* Deprecate `udp=` param
* Add support for `mode=strict` to NL app, since that exercises the `api/data` flow
* Move tests from `explore_test.py` to `nl_test.py`, also for the same reason.

### CODE REVIEW

File: nl_server/routes.py [hunk 1]
```diff
@@ -45,12 +46,23 @@ def search_sv():
   idx = str(escape(request.args.get('idx', config.DEFAULT_INDEX_TYPE)))
   if not idx:
     idx = config.DEFAULT_INDEX_TYPE
+
+  threshold = escape(request.args.get('threshold'))
+  if threshold:
+    try:
+      threshold = float(threshold)
+    except Exception:
```
Comment (shifucun) [inline nl_server/routes.py:54]: add a logging here?

File: server/lib/explore/params.py [hunk 2]
```diff
@@ -31,8 +33,6 @@ class Params(str, Enum):
   # Indicating it's a test query and the type of test, ex, "test=screenshot"
   TEST = 'test'
   I18N = 'i18n'
-  # Whether to use default place when no places are detected from the query.
-  USE_DEFAULT_PLACE = 'udp'
   # The mode of query detection.
   # - 'strict': detect and fulfill query with much higher specificity criteria.
```
Comment (shifucun) [inline server/lib/explore/params.py:37]: in comment also mention no default place?

File: server/lib/nl/common/commentary.py [hunk 3]
```diff
@@ -139,7 +139,7 @@ def user_message(uttr: Utterance) -> UserMessage:
   if (uttr.rankedCharts and
       uttr.sv_source == FulfillmentResult.CURRENT_QUERY and
       get_top_sv_score(uttr.detection, uttr.rankedCharts[0])
-      < LOW_CONFIDENCE_SCORE_REPORT_THRESHOLD):
+      < SV_SCORE_HIGH_CONFIDENCE_THRESHOLD):
```
Comment (shifucun) [inline server/lib/nl/common/commentary.py:142]: why using a high confidence score here? add comment?

File: server/lib/nl/detection/utils.py [hunk 4]
```diff
@@ -29,20 +29,18 @@
...
+def filter_svs(detection: SVDetection, counters: ctr.Counters) -> List[str]:
   i = 0
   ans = []
   blocked_vars = set()
-  while (i < len(sv.svs)):
-    if (sv.scores[i] >= _SV_THRESHOLD):
-      var = sv.svs[i]
+  single_sv = detection.single_sv
+  while (i < len(single_sv.svs)):
+    if (single_sv.scores[i] >= detection.sv_threshold):
```
Comment (shifucun) [inline server/lib/nl/detection/utils.py:42]: nit: no need for extra () in `while` and `if`

File: static/src/server.ts [hunk 5]
```diff
@@ -465,9 +465,9 @@ app.get("/nodejs/query", (req: Request, res: Response) => {
   const urlRoot = `${req.protocol}://${req.get("host")}`;
   res.setHeader("Content-Type", "application/json");
   axios
-    // Use "udp=false" to disable using default place.
+    // Use "mode=strict" to disable using default place, use a higher SV threshold, etc.
     .post(
-      `${CONFIG.apiRoot}/api/nl/data?q=${query}&detector=heuristic&udp=false&mode=strict`,
+      `${CONFIG.apiRoot}/api/nl/data?q=${query}&detector=heuristic&mode=strict`,
```
Comment (chejennifer) [inline static/src/server.ts:470]: nit: from the description in the pr, `mode=strict` also sets `detector=heuristic` right? so we can remove that param?

File: static/js/apps/explore/app.tsx [hunk 6]
```diff
@@ -261,7 +261,6 @@ export function App(props: { isDemo: boolean }): JSX.Element {
     const llmApi = getSingleParam(hashParams[URL_HASH_PARAMS.LLM_API]);
     const testMode = getSingleParam(hashParams[URL_HASH_PARAMS.TEST_MODE]);
     const i18n = getSingleParam(hashParams[URL_HASH_PARAMS.I18N]);
-    const udp = getSingleParam(hashParams[URL_HASH_PARAMS.USE_DEFAULT_PLACE]);
```
Comment (chejennifer) [inline static/js/apps/explore/app.tsx:264]: can probably also remove `USE_DEFAULT_PLACE` from `URL_HASH_PARAMS` right?

File: nl_server/tests/custom_embeddings_test.py [hunk 7]
```diff
@@ -97,7 +97,7 @@ def test_queries(self, dc, query, index, expected):
     if idx:
       got = idx.detect_svs(query)
       for i in range(len(got['SV'])):
-        if got['CosineScore'][i] >= embeddings._SV_SCORE_THRESHOLD:
+        if got['CosineScore'][i] >= 0.5:
```
Comment (chejennifer) [inline nl_server/tests/custom_embeddings_test.py:100]: should this use the `_SV_SCORE_DEFAULT_THRESHOLD`?

**Review by chejennifer — APPROVED** [review]
nice! Just some nits

**Review by pradh — COMMENTED** [review]
Thank you both for the reviews!

**Review by shifucun — APPROVED** [review]

**Pull request merged**